src/utils.js

```diff
@@ -41,10 +42,19 @@ export function loadJSON(file, mappings) {
   }
 }
 
-export function dumpJSON(file, mappings) {
+export function dumpJSON(file, mappings, persistMappings) {
   try {
     log.info(`Writing ${file}`);
-    const jsonBody = JSON.stringify(mappings, null, 2);
+    let jsonBody = JSON.stringify(mappings, null, 2);
+
+    // merge both config if PERSIST_MAPPINGS=true
+    if (persistMappings && fs.existsSync(file)) {
+      const currentConfig = fs.readJSONSync(file);
+      const newConfig = JSON.parse(jsonBody);
+      jsonBody = { ...currentConfig, ...newConfig };
```

Comment:
The merge appears to be naive and overriding at the top level of the JSON object.
If both the local configuration (`currentConfig`) and management API configuration (`newConfig`) are changing fields, we might lost the local configuration changes.

For example, 
```json
currentConfig = {
    "a1": {
        "b1": "current_b1_val",
        "b2": "newer_b2_val"
   }
}

newConfig = {
    "a1": {
        "b1": "newer_b1_val",
        "b2": "current_b2_val"
   }
}

jsonBody = {
    "a1": {
        "b1": "newer_b1_val",
        "b2": "current_b2_val"
   }
}
```

Reply:
@shushen local changes are by default overridden whenever we export the changes from upstream. In that case, also all the fields are overridden by the new params as well as the substitute expression as well which is not somewhat user will prefer to write every time he syncs the changes from upstream.

Reply:
There is no way to keep the mappings user has written once and then wants to update for eg: **stag** tenant by exporting the new configuration. By direct merging, the user can keep its configuration in git and later on do a diff and merge if he thinks some fields are unwanted.

Reply:
Hi @shushen,

There is no way for the user to keep the mapping that he has written once for all the environments. Using git and then merging configurations will allow the user to select the fields which are merged using the above strategy rather than writing those expressions every time configurations are synced from upstream.